src/main/java/org/wso2/maven/humantask/artifact/util/FileUtils.java

```diff
@@ -0,0 +1,168 @@
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import org.apache.log4j.Logger;
+import org.apache.maven.plugin.logging.Log;
+
+public class FileUtils {
+
+	private static final Logger logger = Logger.getLogger(FileUtils.class);
+
+	public static final String ERROR_CREATING_CORRESPONDING_ZIP_FILE = "Error creating corresponding ZIP file";
+
+	public static void copyDirectory(File srcPath, File dstPath, List<File> filesToBeCopied) throws IOException {
+
+		if (srcPath.isDirectory()) {
+			if (!dstPath.exists()) {
+				dstPath.mkdir();
+			}
+			String files[] = srcPath.list();
+			if (files != null) {
+				for (String file : files) {
+					copyDirectory(new File(srcPath, file), new File(dstPath, file), filesToBeCopied);
+				}
+			}
+		} else {
+			if (srcPath.exists()) {
+				org.apache.commons.io.FileUtils.copyFile(srcPath, dstPath);
+			}
+		}
+	}
+
+	public static List<File> getAllFilesPresentInFolder(File srcPath) {
+		List<File> fileList = new ArrayList<File>();
+		if (srcPath.isDirectory()) {
+			String files[] = srcPath.list();
+			if (files != null) {
+				for (String file : files) {
+					fileList.addAll(getAllFilesPresentInFolder(new File(srcPath, file)));
+				}
+			}
+		} else {
+			fileList.add(srcPath);
+		}
+		return fileList;
+	}
+
+	public static File createArchive(Log log, File location, File artifactLocation, String artifactName)
+			throws Exception {
+		List<File> allFilesPresentInFolder = getAllFilesPresentInFolder(artifactLocation);
+		File[] fileArray = new File[allFilesPresentInFolder.size()];
+		for (int i = 0; i < allFilesPresentInFolder.size(); i++) {
+			fileArray[i] = allFilesPresentInFolder.get(i);
+			log.info("FileArray name : " + allFilesPresentInFolder.get(i).getName());
+		}
+		List<File> validFileList = getFileList(fileArray);
+		if (validFileList.size() == 0) {
+			throw new Exception("The selected location " + location.getName() + "(" + location.toString()
+					+ ") does not contain any human task files.");
+		}
+		File targetFolder;
+		targetFolder = new File(location.getPath(), "target");
+		File humantaskDataFolder = new File(targetFolder, "ht-tmp");
+		humantaskDataFolder.mkdirs();
+		File zipFolder = new File(humantaskDataFolder, artifactLocation.getName());
+		zipFolder.mkdirs();
+		copyDirectory(artifactLocation, zipFolder, validFileList);
+		log.info("Copied Size : " + getAllFilesPresentInFolder(zipFolder).size());
+		File zipFile = new File(targetFolder, artifactName);
+		zipFolder(zipFolder.getAbsolutePath(), zipFile.toString());
+		org.apache.commons.io.FileUtils.deleteDirectory(humantaskDataFolder);
+		return zipFile;
+
+	}
+
+	public static List<File> getFileList(File[] fileList) {
+		List<File> list = new ArrayList<File>();
+		Collections.addAll(list, fileList);
+		return list;
+	}
+
+	static public void zipFolder(String srcFolder, String destZipFile) {
+		try (FileOutputStream fileWriter = new FileOutputStream(destZipFile);
```

Comment:
Try with resources method is used to close the stream here. Fixed in https://github.com/wso2/devstudio-tooling-bps/pull/25